[s] My best friend is a holocarp

coiax opened the pull request:
So you know what would be better than boring holoparasites? HOLOCARP!

Just put the holocarp fishsticks in your mouth, and you'll be able to catch your own holocarp!

Sure, you can't pick the type of them, but you can have more than one to compensate. Is this balanced? Who knows?

Bonus points if you recognise the fish name list.

coiax committed:
Time for the holocarp realness

Summon your holocarp with your fish sticks. Collect them all!

coiax committed:
Carp names

Bonus points if you get the reference. There's a reason it's not
alphabetically sorted.

coiax committed:
HOLOCARP ENGAGE

At some point, we'll run out of fish related jokes.

Anyway, put terrible carp descriptions of each of the classes, and changed
the icon of a carp themed guardian to a holocarp. Doesn't change colour.

coiax committed:
Hooker Green reads better than Hooker's

coiax committed:
Apparently quotation marks matter

Jordie0608 commented:
I find this humorous

Jordie0608 commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -40,10 +40,18 @@ var/global/list/parasites = list() //all currently existing/living guardians
 	setthemename(theme)
+
+	//Special case holocarp, because #snowflake code
+	if(theme == "carp")
```
This would be much better if you made a child of the guardian mob and called it during its New()

ChangelingRain replied:
Can't be a child because of all the types; it'd be better to add these(icon/emote/desc) to the theme datum and setthemename()
Which would allow for a lot more parasite types, too.

Supermichael777 commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -52,7 +60,7 @@ var/global/list/parasites = list() //all currently existing/living guardians
 /mob/living/simple_animal/hostile/guardian/proc/setthemename(pickedtheme) //set the guardian's theme to something cool!
 	if(!pickedtheme)
-		pickedtheme = pick("magic", "tech")
+		pickedtheme = pick("magic", "tech", "carp")
```
this will make it randomly pick if no theme rather then a preselected theme from the device. you didn't need to touch it.

Supermichael777 replied:
this will make it randomly pick if no theme rather then a preselected theme from the device. you didn't need to touch it.

ChangelingRain replied:
It's not necessary but this pick shouldn't be called anyway; it should always be getting a specific theme unless you're doing testing and have disabled dying without a summoner

ChangelingRain commented:
This is hilarious and I love it.

MMMiracles commented:
https://youtu.be/eaCCkfjPm0o?t=4m9s

i recognized this way too quickly

coiax committed:
Moved snowflaking the carp theme to updatetheme()

Based on feedback, the changes are now done in updatetheme().

Also with additional biting noises.

coiax commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -83,6 +77,16 @@ var/global/list/parasites = list() //all currently existing/living guardians
 	icon_dead = "[namedatum.parasiteicon]"
 	bubble_icon = "[namedatum.bubbleicon]"
 
+	//Special case holocarp, because #snowflake code
```
I moved the special casing to updatetheme() as suggested.

coiax committed:
Holocarp now dye themselves the colour

To be used with the greyscale sprite provided.

ChangelingRain commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -61,10 +63,13 @@ var/global/list/parasites = list() //all currently existing/living guardians
+	updatetheme(pickedtheme)
 
-/mob/living/simple_animal/hostile/guardian/proc/updatetheme() //update the guardian's theme to whatever its datum is; proc for adminfuckery
+/mob/living/simple_animal/hostile/guardian/proc/updatetheme(theme) //update the guardian's theme to whatever its datum is; proc for adminfuckery
```
You don't actually need to pass the theme for this; you can do `if(namedatum.theme == "carp")` instead

Bawhoppen commented:
Uh you didn't make this to the april fools branch... dear god I hope it's not intentional

coiax commented:
Oh, yeah, whoops. (To be honest, I'm not sure it's that terrible a change, given at present, holocarp fishsticks would be admin spawn only.)

Would I have to  close this and open a new one?

duncathan commented:
yea, you'll have to remake the PR.
